rs/consensus/src/consensus/batch_delivery.rs

```diff
@@ -294,78 +295,104 @@ pub fn generate_responses_to_subnet_calls(
 /// computed DKG key material for remote subnets, without needing values from the state.
-pub fn generate_responses_to_setup_initial_dkg_calls(
+pub fn generate_responses_to_remote_dkgs(
     transcripts_for_remote_subnets: &[(NiDkgId, CallbackId, Result<NiDkgTranscript, String>)],
     log: &ReplicaLogger,
 ) -> Vec<ConsensusResponse> {
-    let mut consensus_responses = Vec::new();
-
-    let mut transcripts: BTreeMap<CallbackId, TranscriptResults> = BTreeMap::new();
-
+    let mut dkg_results: BTreeMap<CallbackId, RemoteDkgResults> = BTreeMap::new();
     for (id, callback_id, transcript) in transcripts_for_remote_subnets.iter() {
-        let add_transcript = |transcript_results: &mut TranscriptResults| {
-            let value = Some(transcript.clone());
-            match &id.dkg_tag {
-                NiDkgTag::LowThreshold => {
-                    if transcript_results.low_threshold.is_some() {
-                        error!(
-                            log,
-                            "Multiple low threshold transcripts for {}", callback_id
-                        );
-                    }
-                    transcript_results.low_threshold = value;
-                }
-                NiDkgTag::HighThreshold => {
-                    if transcript_results.high_threshold.is_some() {
-                        error!(
-                            log,
-                            "Multiple high threshold transcripts for {}", callback_id
-                        );
-                    }
-                    transcript_results.high_threshold = value;
-                }
-                NiDkgTag::HighThresholdForKey(master_public_key_id) => {
-                    /////////////////////////////////////
-                    // TODO(CON-1416): Generalize this function to support both SetupInitialDKG and vetKD key resharing
-                    /////////////////////////////////////
-                    error!(
-                        log,
-                        "Implementation error: NiDkgTag::HighThresholdForKey({master_public_key_id}) used in SetupInitialDKG for callback ID {callback_id}",
-                    );
-                }
-            }
-        };
-        match transcripts.get_mut(callback_id) {
-            Some(existing) => add_transcript(existing),
-            None => {
-                let mut transcript_results = TranscriptResults {
-                    low_threshold: None,
-                    high_threshold: None,
-                };
-                add_transcript(&mut transcript_results);
-                transcripts.insert(*callback_id, transcript_results);
-            }
-        };
+        dkg_results
+            .entry(*callback_id)
+            .and_modify(|transcript_result| {
+                transcript_result.add_transcript(id, transcript.clone(), log)
+            })
+            .or_insert(RemoteDkgResults::new(id, transcript.clone()));
```

Comment:
nit: in case you want to avoid cloning the transcript a second time if the entry already exists
```suggestion
            .or_insert_with(|| RemoteDkgResults::new(id, transcript.clone()));
```